Extensions/xkit_patches.js

```diff
@@ -223,6 +223,14 @@ XKit.tools.dump_config = function(){
 	}, {});
 };
 
+// http://stackoverflow.com/a/901144/2073440
+XKit.tools.getParameterByName = function(name){
+	name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+	var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
```

Comment:
It escapes the first set of brackets so that it meets the [URI format listed in RFC 3986](http://stackoverflow.com/a/11490435). For example, it will convert `url[123]` into `url\[123\]`.

But on closer inspection:

- it escapes only the first set of brackets, so a string like `url[123][456]` would turn into `url\[123\][456]`
- it escapes via backslashes, not percent encoding
- [as you mentioned](https://github.com/new-xkit/XKit/pull/832#issuecomment-173037201), in this particular case the string is already encoded so chances are it won't find the bracket anyway

_(Note to self: don't blindly copy functions from StackOverflow. Note to other developers: don't be me.)_

Assuming `XKit.tools.getParameterByName` will be a util function for generic use, we have the following options:

- accept this as is
- find an alternative that percent-encodes brackets
- depending on how much we expect this case, find a solution that handles getting the parameter `x` from a URL like `http://www.mysite.com/index.php?x=x1&x=x2&x=x3`
  - considering this deeply may bring us down the rabbit hole of creating bulletproof util functions, or functions that handle every case thrown at them
  - depends on what we think is "good enough"
- something else

What do you suggest?